# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../root/namespace-imp-fix.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- root/utils/src/rootcint.cxx~	2005-06-14 01:26:45.000000000 +0200
+++ root/utils/src/rootcint.cxx	2005-09-02 21:09:45.000000000 +0200
@@ -2306,7 +2306,10 @@
    fprintf(fp, "   namespace ROOT {\n");
 
 #if !defined(R__SGI)
-   fprintf(fp, "      inline ::ROOT::TGenericClassInfo *GenerateInitInstance();\n");
+   if (nesting)
+     fprintf(fp, "      ::ROOT::TGenericClassInfo *GenerateInitInstance();\n");
+   else
+     fprintf(fp, "      inline ::ROOT::TGenericClassInfo *GenerateInitInstance();\n");
 #endif
 
    if (!cl.HasMethod("Dictionary") || cl.IsTmplt())
@@ -2317,7 +2320,10 @@
    fprintf(fp, "      // Function generating the singleton type initializer\n");
 
 #if !defined(R__SGI)
-   fprintf(fp, "      inline ::ROOT::TGenericClassInfo *GenerateInitInstance()\n      {\n");
+   if (nesting)
+     fprintf(fp, "      ::ROOT::TGenericClassInfo *GenerateInitInstance()\n      {\n");
+   else
+     fprintf(fp, "      inline ::ROOT::TGenericClassInfo *GenerateInitInstance()\n      {\n");
 #else
    fprintf(fp, "      ::ROOT::TGenericClassInfo *GenerateInitInstance()\n      {\n");
 #endif
